### PR TITLE
feat: Save the original h5ad as an artifact in the dataset.

### PR DESCRIPTION
**Pull request by Bento007**
- #2124

### Reviewers
**Functional:** @nayib-jose-gloria 

**Readability:** @danieljhegeman 

---

## Changes
- Add saving of the original h5ad file. The original h5ad is named raw.h5ad in the dataset_artifact.filename field. This adds a second H5AD artifact to the dataset. This artifact will be filtered out from responses to users.
- Add filenames to CorporaConstants
- Upgrade sqlalchemy version so all services are using the same version.
- Storing the original uploaded file
- Removing the original upload files when converting ORM object to dict so it doesn't show up in API responses.

### CODE REVIEW

File: backend/corpora/dataset_processing/process_download_validate.py [hunk 1]
```diff
@@ -54,6 +54,9 @@ def process(dataset_id: str, dropbox_url: str, artifact_bucket: str):
 
     # Upload the labeled dataset to the artifact bucket
     bucket_prefix = get_bucket_prefix(dataset_id)
+    create_artifact(
```
Comment (nayib-jose-gloria) [inline backend/corpora/dataset_processing/process_download_validate.py:57]: nit: mind adding a comment (or adding to the comment right above) to explain that we're saving both the unlabeled and labeled h5ads?

**Review by nayib-jose-gloria — APPROVED** [review]
one non-blocking nit to clarify a comment, but looks good!

File: tests/unit/backend/corpora/api_server/test_v1_dataset.py [hunk 2]
```diff
@@ -315,7 +316,13 @@ def test__get_dataset_assets(self):
             user_submitted=True,
             s3_uri="s3://mock-bucket/mock-key.h5ad",
         )
-        dataset = self.generate_dataset(self.session, id="test_dataset", artifacts=[artifact_0, artifact_1])
+        artifact_2 = dict(
+            filename=CorporaConstants.ORIGINAL_H5AD_ARTIFACT_FILENAME,
+            filetype=DatasetArtifactFileType.H5AD,
+            user_submitted=True,
+            s3_uri="s3://mock-bucket/mock-key.h5ad",
```
Comment (nayib-jose-gloria) [inline tests/unit/backend/corpora/api_server/test_v1_dataset.py:323]: for clarity on which artifact should be found in assertions below, we should rename this s3_uri to something else (right now its matching artifact_1)

**Review by nayib-jose-gloria — COMMENTED** [review]
One test set-up change requested, otherwise looks good!

**Review by nayib-jose-gloria — APPROVED** [review]
ty!

**Comment by atolopko-czi** [comment]
@Bento007 Realizing this is already merged, but would it worth be splitting `datasetartifactfiletype` psql enum value `H5AD` into `H5AD_RAW` and `H5AD_LABELED`?

**Pull request merged**